src/components/manage/Widgets/ImageWidget.jsx

```diff
@@ -0,0 +1,324 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useLocation } from 'react-router-dom';
+import {
+  flattenToAppURL,
+  getBaseUrl,
+  isInternalURL,
+  usePrevious,
+} from '@plone/volto/helpers';
+import { FormFieldWrapper, Icon } from '@plone/volto/components';
+import { Button, Dimmer, Input, Loader, Message } from 'semantic-ui-react';
+import { useSelector, useDispatch } from 'react-redux';
+import loadable from '@loadable/component';
+import { defineMessages, useIntl } from 'react-intl';
+import { createContent } from '@plone/volto/actions';
+import { readAsDataURL } from 'promise-file-reader';
+import withObjectBrowser from '@plone/volto/components/manage/Sidebar/ObjectBrowser';
+import config from '@plone/volto/registry';
+
+import imageBlockSVG from '@plone/volto/components/manage/Blocks/Image/block-image.svg';
+import clearSVG from '@plone/volto/icons/clear.svg';
+import navTreeSVG from '@plone/volto/icons/nav.svg';
+import aheadSVG from '@plone/volto/icons/ahead.svg';
+import uploadSVG from '@plone/volto/icons/upload.svg';
+
+const Dropzone = loadable(() => import('react-dropzone'));
+
+const messages = defineMessages({
+  ImageBlockInputPlaceholder: {
+    id: 'Browse the site, drop an image, or type an URL',
+    defaultMessage: 'Browse the site, drop an image, or type an URL',
+  },
+});
+
+const ImageWidget = (props) => {
+  const {
+    block,
+    id,
+    inline,
+    onChange,
+    openObjectBrowser,
+    onSelectBlock,
+    value,
+  } = props;
+  const intl = useIntl();
+  const pathname = useLocation().pathname;
+  const placeholder =
+    props?.data?.placeholder ||
+    props.placeholder ||
+    intl.formatMessage(messages.ImageBlockInputPlaceholder);
+
+  const [url, setUrl] = useState('');
+  const [dragging, setDragging] = useState(false);
+  const [uploading, setUploading] = useState(false);
+  let loading = '';
+  const dispatch = useDispatch();
+
+  const request = useSelector((state) => state.content.subrequests[block]);
+  const content = request?.data;
+  const urlUploaded = content ? content['@id'] : null;
+  const requestLoaded = request ? request.loaded : null;
+
+  useEffect(() => {
+    if (loading) {
+      if (loading && requestLoaded && uploading) {
```

Comment:
Duplicate `loading` check. Removing the first one should be just fine.

Reply:
done.